doc/whats_new.rst

```diff
@@ -251,6 +252,12 @@ API changes summary
       :func:`sklearn.model_selection.cross_val_predict`.
       :issue:`2879` by :user:`Stephen Hoover <stephen-hoover>`.
 
+    - In :class:`pipeline.Pipeline`, ``steps`` has been changed by ``steps`` and
```

Comment:
I think that we should mention in the warning and the what's new when we intend to end the backwards-compatible behaviour. 0.22 at least.